File: Semestr 1/TablicaChar/Szyfr_Cezara.py
"""
autor: Paweł Sadowski
Program służący do szyfrowania i deszyfrowania tekstu za pomocą szyfru Cezara.
"""


x = int(input("Podaj 1 (szyfrowanie) lub 2 (deszyfrowanie): "))
y = int(input("Podaj wartość przesunięcia (klucz): "))
slowo = input("Podaj tekst: ")


def szyfr_cezara(wyraz, przesuniecie):
    wynik = ""
    for znak in wyraz:
        if znak.isalpha():
            if znak.islower():
                baza = ord('a')
            else:  # Wielkie litery (A-Z)
                baza = ord('A')
            nowy_znak = chr((ord(znak) - baza + przesuniecie) % 26 + baza)
            wynik += nowy_znak
        else:
            wynik += znak
    return wynik


if x == 1:
    print("Szyfrowanie...")
    zaszyfrowany = szyfr_cezara(slowo, y)
    print(f"Zaszyfrowany tekst: {zaszyfrowany}")
elif x == 2:
    print("Deszyfrowanie...")
    odszyfrowany = szyfr_cezara(slowo, -y)
    print(f"Odszyfrowany tekst: {odszyfrowany}")
else:
    print("Nieprawidłowa opcja. Wybierz 1 lub 2.")

File: Semestr 1/TablicaChar/Szyfr_Cezara.cpp
/*
autor: Pawel Sadowski
Program s³u¿¹cy do szyfrowania i deszyfrowania tzw szyfru cezara

*/

#include <iostream>
#include <cstring>

using namespace std;

int main()
{

    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    short x,k;
    char tmp[10001];
    cin >> x >> k;
    cin.ignore();
    cin.clear();
    cin.getline(tmp,10001);
    int dlugosc = strlen(tmp);

    if(x == 1)
    {
        for (int i = 0; i < dlugosc; i++)
        {
            for (int j = 0; j < k; j++)
            {
                if (tmp[i] >= 'a' && tmp[i] <= 'z')
                {
                    if (tmp[i] == 'z')
                        tmp[i] = 'a';
                    else
                        tmp[i]++;
                }
                else if (tmp[i] >= 'A' && tmp[i] <= 'Z')
                {
                    if (tmp[i] == 'Z')
                        tmp[i] = 'A';
                    else
                        tmp[i]++;
                }
            }
        }
    }
    else
    {
        for (int i = 0; i < dlugosc; i++)
        {
            for (int j = 0; j < k; j++)
            {
                if (tmp[i] >= 'a' && tmp[i] <= 'z')
                {
                    if (tmp[i] == 'a')
                        tmp[i] = 'z';
                    else
                        tmp[i]--;
                }
                else if (tmp[i] >= 'A' && tmp[i] <= 'Z')
                {
                    if (tmp[i] == 'A')
                        tmp[i] = 'Z';
                    else
                        tmp[i]--;
                }
            }
        }
    }
    for(int i=0; i < dlugosc; i++)
        cout << tmp[i];
    return 0;
}
